celery/app/defaults.py

```diff
@@ -95,6 +95,8 @@ def __repr__(self):
         heartbeat=Option(120, type='int'),
         heartbeat_checkrate=Option(3.0, type='int'),
         login_method=Option(None, type='string'),
+        native_delayed_delivery=Option(False, type='bool'),
```

Comment:
That's what the warning is for. We need this feature flag so that this can be turned on manually when all those exchanges and queues are declared only when needed.
People may chose to leave this turned off for their own reasons.

Reply:
Related to: https://github.com/celery/celery/pull/9207/files#r1801792448

Reply:
> People may choose to leave this turned off for their own reasons.

Give me one example that it is useful as `False`.
Any case anyone can benefit from it?

I argue that the `False` path is not useful for any use-case, which is why it should be `True`, or actually not even used and assumed `True` automatically if quorum queues are detected, per no use case at all of the `False` option.

Reply:
You may want to keep the old, timer based solution when not using quorum queues. Hence, the sensible default is `False`.
You may also want to use native delayed delivery without quorum queues, in which case the flag should be set to `True`. That's exactly what my client wants to do first.
They want to test this feature without using quorum queues first and later on migrate to quorum queues.